oak_idl/src/lib.rs

```diff
@@ -20,120 +20,39 @@ extern crate alloc;
 
 use alloc::vec::Vec;
 use core::fmt::Debug;
-use serde::{Deserialize, Serialize};
 
 pub mod utils;
 
 #[derive(Debug)]
-pub enum ClientError {
+pub enum Error {
+    /// The request message could not be deserialized correctly.
     InvalidRequest,
+    /// The response message could not be deserialized correctly.
     InvalidResponse,
-    TransportError(TransportError),
-}
-
-impl From<TransportError> for ClientError {
-    fn from(e: TransportError) -> Self {
-        ClientError::TransportError(e)
-    }
-}
-
-#[derive(Debug)]
-pub enum TransportError {
-    InvalidMessage,
-    InvalidTransactionId,
+    /// The method id provided for the invocation was not implemented by the server.
     InvalidMethodId,
 }
 
 /// Unique identifier of a method within a service.
 type MethodId = u32;
 
-/// Header of a transactional message.
-///
-/// See <https://fuchsia.dev/fuchsia-src/reference/fidl/language/wire-format#transactional-messages>
-#[derive(Serialize, Deserialize)]
-pub struct Header {
-    /// Identifies the transaction over the underlying transport. Each transaction (a.k.a. call or
-    /// invocation) over the same transport must have a unique transaction id (regardless of the
-    /// method_id value).
-    ///
-    /// The client picks a unique transaction id to put in the request header, and when the server
-    /// replies to that message, it also uses the same transaction id in the response message.
-    pub transaction_id: u32,
-
+/// A request message representing an invocation.
```

Comment:
Do I understand correctly: A `Request` to invoke a method identified by `method_id` with the arguments given in `body` ?